openff/toolkit/topology/molecule.py

```diff
@@ -5827,30 +5847,11 @@ def get_bond_between(self, i, j):
 
         raise NotBondedError("No bond between atom {} and {}".format(i, j))
 
-    @property
-    def rings(self):
```

Comment:
In its current state, this PR removes `Molecule.rings` without replacing it.

Now that I know more about the area, I don't know that we should have exposed `Molecule.rings` and `Molecule.n_rings` in the first place (because, as indicated in both OE and RDKit docs, there are multiple distinct valid behaviors)

If we keep both methods and let them use either the OE or RDK backends, the docstring should have a big warning that different backends will give different outputs. I'm kinda concerned that supporting both backends would invite further scope creep ("if we read into the options we may be able to make them more consistent") and violate the principle of "OpenFF isn't a cheminformatics toolkit". 

On the other hand, if we deprecate `Molecule.rings` and `n_rings`, we could still offer `Atom.is_in_ring` and `Bond.is_in_ring` (which have unambiguous behavior, thereby satisfying @SimonBoothroyd's needs), having each call out directly to a toolkit. In the release notes, we could say "hey, if you need rings, you'll need to do it directly through a cheminformatics toolkit now".

I'm kinda in favor of the "deprecate" route. Thoughts?

Reply:
+1 of just deprecating then removing `rings` and `n_rings`. It would also be good to consider allowing `is_in_ring` to take an 'of_size' argument if it's being changed into a function. This would allow us to more easily encode 'is in ring of size' into a feature vector using the toolkit!